### PR TITLE
docs(CONTRIBUTING): add “Local development”

### LINKED ISSUE #91: Documentation: prove contributing page adding step by step guide to setup the project
I'm setting up the project to work on #84.

I cloned the repository and I have run `pnpm install`.
After that I have encountered 2 problems:

### 1. `apps/documentation` types not resolved on VSCode
on `apps/documentation` files I'm not able to have import resolved correctly in the editor:

vs-code-resolve-errors

> [!NOTE]
> When running `lint` or `types` task from the `CLI` no error is present.
> I can work by running CLI commands but I found out a little strange that the editor is not behaving the same.
> It might be a VSCode issue, but I'm just wondering if you have a local setup or something else I might missed that can solve the issue.

### 2. `dev` task loop

Running `pnpm dev`  causes a loop:

```text
tuono-fs-router-vite-plugin:dev: 
tuono-fs-router-vite-plugin:build: cache miss, executing 552cb6325ef9d96f
tuono-fs-router-vite-plugin:dev: cache miss, executing 4361ae84ef2c11b2
tuono-fs-router-vite-plugin:build: 
tuono-fs-router-vite-plugin:dev: 
tuono-fs-router-vite-plugin:dev: 
tuono-fs-router-vite-plugin:dev: > tuono-fs-router-vite-plugin@0.12.0 dev /Users/marcalexiei/development/tuono/packages/fs-router-vite-plugin
tuono-fs-router-vite-plugin:dev: > vite build --watch
tuono-fs-router-vite-plugin:dev: 
tuono-fs-router-vite-plugin:build: 
tuono-fs-router-vite-plugin:build: > tuono-fs-router-vite-plugin@0.12.0 build /Users/marcalexiei/development/tuono/packages/fs-router-vite-plugin
tuono-fs-router-vite-plugin:build: > vite build
tuono-fs-router-vite-plugin:build: 
tuono-fs-router-vite-plugin:dev: cache miss, executing 87911e2c75a3e2d2
tuono-fs-router-vite-plugin:build: cache miss, executing 383aa3afd2063cef
```

<details>
<summary>Logs screenshot</summary>

build loop screenshot

</details>

---

### System information

```text
  System:
    OS: macOS 15.1
    CPU: (8) arm64 Apple M3
    Memory: 88.48 MB / 16.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 22.11.0 - ~/.nvm/versions/node/v22.11.0/bin/node
    npm: 10.9.0 - ~/.nvm/versions/node/v22.11.0/bin/npm
    pnpm: 9.12.3 - ~/.nvm/versions/node/v22.11.0/bin/pnpm
```

---

It might be worth to create a step by step setup guide on the [contributing page of the documentation site](https://tuono.dev/documentation/contributing) with the list of the commands to setup all dependencies, build and run tests to check that everything in the project is working properly? 
Since you have told me you are working on the server implementation when we resolve these issues I can take care of it.

### PR DESCRIPTION
**Pull request by marcalexiei**
## Context & Description

Close #91.

<details>

<summary>Page preview</summary>

Screenshot 2024-11-22 at 12 50 19

</details>

### Consideration

if you find the page too long, we could consider to split it into 3 different pages:
- setup
- development
- documentation

### CODE REVIEW

File: apps/documentation/src/routes/documentation/contributing/local-development.mdx [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+1. Start tuono frontend build using
+
+   ```sh
+   turbo run dev
+   ```
+
+1. In another shell run
+
+   ```sh
+   cargo build
```
Comment (Valerioageno) [inline apps/documentation/src/routes/documentation/contributing/local-development.mdx:83]: We could mention to install `cargo-watch` crate that allows automatic re-builds with `cargo watch -x build -w crates/`

File: apps/documentation/src/routes/documentation/contributing/local-development.mdx [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+<MetaTags
+  title="Tuono - Contributing - Local development"
+  canonical="https://tuono.dev/documentation/contributing/local-development"
+  description="The project is massive - if you like it, do consider contributing!"
+/>
+
+import Breadcrumbs, { Element } from '../../../components/breadcrumbs'
+
+<Breadcrumbs breadcrumbs={[{ label: '✨ Contributing' }]} />
```
[the remark this answers is not in the document]
Comment (marcalexiei) [reply apps/documentation/src/routes/documentation/contributing/local-development.mdx]: Copy-paste strikes again!

**Comment by Valerioageno** [comment]
How about calling the page `Repository setup`?

File: apps/documentation/src/routes/documentation/contributing/local-development.mdx [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+<Breadcrumbs
+  breadcrumbs={[
+    { label: '✨ Contributing', href: '/documentation/contributing' },
+    { label: 'Local development' },
+  ]}
+/>
+
+# Local development
+
+Thanks for your interest! In this page, you find the instructions to set up the `tuono`!
```
Comment (Valerioageno) [inline apps/documentation/src/routes/documentation/contributing/local-development.mdx:20]:
```suggestion
Thanks for your interest! In this page, you find the instructions to set up `tuono` on your local environment!
```

File: apps/documentation/src/routes/documentation/contributing/local-development.mdx [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+
+## Quick setup
```
Comment (Valerioageno) [inline apps/documentation/src/routes/documentation/contributing/local-development.mdx:22]: I don't get the benefit of having both "Setup" and "Quick Setup". 
Aren't we giving the same value twice?

File: apps/documentation/src/routes/documentation/contributing/local-development.mdx [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+
+### NodeJS — package manager
+
+We use [`pnpm`](https://pnpm.io) as `NodeJS` package manager.
+
+You can see which version of yarn we use by checking the `packageManager` field in the root `package.json`.
+
+### Pre-flight checks
+
+Check that everything is working properly run:
```
Comment (Valerioageno) [inline apps/documentation/src/routes/documentation/contributing/local-development.mdx:70]:
```suggestion
To check that everything is working properly run:
```

File: apps/documentation/src/routes/documentation/contributing/local-development.mdx [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+
+   > Consider to add an alias to your shell setup file
+   >
+   > ```sh
+   > alias t="/path-to-repo/target/debug/tuono"
+   > ```
+
+## Documentation development
+
+1. Change the current working directory to the docs folder:
```
Comment (Valerioageno) [inline apps/documentation/src/routes/documentation/contributing/local-development.mdx:107]: Shouldn't this be a different number?

```suggestion
2. Change the current working directory to the docs folder:
```

File: apps/documentation/src/routes/documentation/contributing/local-development.mdx [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+   cd apps/documentation
+   ```
+
+2. Run
+
+   ```sh
+   tuono dev
+   ```
+
+3. Open the localhost URL.
```
Comment (Valerioageno) [inline apps/documentation/src/routes/documentation/contributing/local-development.mdx:119]:
```suggestion
3. Open the localhost URL [http://localhost:3000/](http://localhost:3000).
```

File: apps/documentation/src/routes/documentation/contributing/local-development.mdx (hunk 4 above)
Comment (marcalexiei) [reply apps/documentation/src/routes/documentation/contributing/local-development.mdx:22]: I was thinking about avoiding to have an "advanced user" scroll all the short explanation of each installation. 
But re-reading made me realise it is not worth it.

File: apps/documentation/src/routes/documentation/contributing/local-development.mdx [hunk 8]
```diff
@@ -0,0 +1,113 @@
+import MetaTags from '../../../components/meta-tags'
+
+<MetaTags
+  title="Tuono - Contributing - Local development"
+  canonical="https://tuono.dev/documentation/contributing/local-development"
+  description="The project is massive - if you like it, do consider contributing!"
```
Comment (Valerioageno) [inline apps/documentation/src/routes/documentation/contributing/local-development.mdx:6]:
```suggestion
  description="Contribute to Tuono. Learn here how to setup the repository for local development"
```

**Review by Valerioageno — APPROVED** [review]
LGTM. Just added a very little comment.

**Comment by marcalexiei** [comment]
Switching to draft, I'll try to fix the spell checking error shortly

**Pull request merged**